tests/unit/factored_matrix/test_multiply_by_vector.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+import torch
+from torch.testing import assert_close
+
+from transformer_lens import FactoredMatrix
+
+
+def test_left_matmul_by_vector_left():
+    a = torch.rand(2, 3)
+    b = torch.rand(3, 4)
+
+    fm = FactoredMatrix(a, b)
+    vector = torch.rand(4)
+
+    assert_close(fm @ vector, (a @ b) @ vector)
+
+
+def test_left_matmul_by_vector_leading_dim():
+    a = torch.rand(6, 2, 3)
+    b = torch.rand(6, 3, 4)
+
+    fm = FactoredMatrix(a, b)
+    vector = torch.rand(4)
+
+    assert_close(fm @ vector, (a @ b) @ vector)
+
+
+@pytest.mark.skip(
+    """
+    This test fails with
+    AssertionError: The values for attribute 'shape' do not match: torch.Size([1, 4]) != torch.Size([4]).
+
+    I think this is a bug in __rmatmul__:
+    `return ((other.unsqueeze(-2) @ self.A) @ self.B).squeeze(-1)` is squeezing on the wrong dimension.
+
+    We could fix this by changing the argument to squeeze to `-2`. However there would be a small risk of
+    breaking consumers that assume the existing behaviour?
+    """
```

Comment:
@rusheb I think this is something I'd feel comfortable changing because it feels like a standard bug. 

It would be nice to have the tutorial-regression tests working to for more oversight on whether changes at this level will likely affect many people.

Reply:
Raised issue: will address in a separate PR to so that behaviour changes are easy to roll back! 

https://github.com/neelnanda-io/TransformerLens/issues/225